From f31c4cd950b3b40ceed5392f3eac0ba9e6fa2312 Mon Sep 17 00:00:00 2001
From: root <root@poweramdcruncher5000.labor.koeln.ccc.de>
Date: Sat, 7 Jun 2014 17:19:21 +0200
Subject: [PATCH] Implement a fallback String.prototype.startsWith()

This is scheduled to be in ECMAScript 6.
Right now only Firefox has it.
---
 licht.js |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/licht.js b/licht.js
index 6b0dbf9..1238325 100644
--- a/licht.js
+++ b/licht.js
@@ -1,3 +1,9 @@
+if (typeof String.prototype.startsWith != 'function') {
+    String.prototype.startsWith = function (str){
+        return this.slice(0, str.length) == str;
+    };
+}
+
 function register_click_handlers()
 {
     $('#gate').each(function() {
-- 
1.7.10.4

